File: tree-editor-frontend/local_node_modules/treedocumentmodel/TreeDocument.ts
import {TreeRoot} from "./TreeRoot";
import {OrderedSet} from "crossecore/lib/OrderedSet";
import {InternalEObject} from "crossecore/lib/InternalEObject";

export interface TreeDocument
extends InternalEObject

{
	
	roots: OrderedSet<TreeRoot>;
	

}

File: model-server/src/main/java/no/ntnu/stud/krirek/treelsp/model/tree/TreeDocument.java
package no.ntnu.stud.krirek.treelsp.model.tree;

import org.immutables.gson.Gson;
import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

import java.util.List;

@Value.Immutable @Gson.TypeAdapters
public interface TreeDocument {

    @NotNull List<TreeRoot> roots();
}
